metadata-ingestion/src/datahub/ingestion/source/sql/snowflake.py

```diff
@@ -539,17 +539,38 @@ def do_provision_role_internal(self):
                 [
                     f"grant usage on DATABASE {db_name} to role {role}",
                     f"grant usage on all schemas in database {db_name} to role {role}",
-                    f"grant select on all tables in database {db_name} to role {role}",
-                    f"grant select on all external tables in database {db_name} to role {role}",
-                    f"grant select on all views in database {db_name} to role {role}",
                     f"grant usage on future schemas in database {db_name} to role {role}",
-                    f"grant select on future tables in database {db_name} to role {role}",
                 ]
             )
+            if self.config.profiling.enabled:
+                sqls.extend(
+                    [
+                        f"grant select on all tables in database {db_name} to role {role}",
+                        f"grant select on future tables in database {db_name} to role {role}",
+                        f"grant select on all external tables in database {db_name} to role {role}",
+                        f"grant select on future external tables in database {db_name} to role {role}",
+                        f"grant select on all views in database {db_name} to role {role}",
+                        f"grant select on future views in database {db_name} to role {role}",
+                    ]
+                )
+            else:
+                sqls.extend(
+                    [
+                        f"grant REFERENCES on all tables in database {db_name} to role {role}",
```

Comment:
nitpick: No need to capitalize REFERENCES - can be references to match other statement convention.

Reply:
done